lib/lightning/impact_tracking/worker.ex

```diff
@@ -10,15 +10,33 @@ defmodule Lightning.ImpactTracking.Worker do
     max_attempts: 1
 
   alias Lightning.ImpactTracking.Client
+  alias Lightning.ImpactTracking.Configuration
+  alias Lightning.ImpactTracking.Report
+  alias Lightning.Repo
 
   @impl Oban.Worker
   def perform(_opts) do
     if Application.get_env(:lightning, :impact_tracking)[:enabled] do
+      find_configuration()
+
       host = Application.get_env(:lightning, :impact_tracking)[:host]
 
-      Client.submit_metrics(%{}, host)
+      Client.submit_metrics(%{}, host) |> create_report()
     end
 
     :ok
   end
+
+  defp find_configuration do
+    Configuration |> Repo.one() || %Configuration{} |> Repo.insert()
```

Comment:
I think you meant `Repo.insert!()` to have the same return type. An idiomatic suggestion would be:
```
with nil <- Repo.one(Configuration) do
  Repo.insert!(%Configuration{})
end
```